roles/kubernetes/master/templates/kubeadm-config.v1beta1.yaml.j2

```diff
@@ -151,14 +151,14 @@ apiServer:
   - name: {{ audit_log_name }}
     hostPath: {{ audit_log_hostpath }}
     mountPath: {{ audit_log_mountpath }}
-    writable: true
+    readOnly: false
 {% endif %}
 {% endif %}
 {% for volume in apiserver_extra_volumes %}
   - name: {{ volume.name }}
     hostPath: {{ volume.hostPath }}
     mountPath: {{ volume.mountPath }}
-    writable: {{ volume.writable | default(false)}}
+    readOnly: {{ volume.readOnly | d(not volume.writable) }}
```

Comment:
I think that should include a default for `volume.writable` as well like `volume.readOnly | d(not (volume.writable|d(false)))` in case neither writable nor readonly are defined

Reply:
That's the beauty part, since the `volume.writable` in `d(not volume.writable)` is falsey when writable is not defined, this defaults to `true` which is desired. An additional `default(true)` could be added, but it will never run.

Reply:
That's what I thought at first... I'm sorry to have let my fear of AnsibleUndefinedVariable get the better of me.